### PR TITLE
Fix get_wheel_filename being out of date

### LINKED ISSUE #39842: [runtime env] Code to reproduce wheel download URLs is out of date
The current master wheel download URLs are at 

https://github.com/ray-project/ray/blob/bb892a6bd0224ab3ae11859268ddef01cbb68e71/doc/source/ray-overview/installation.rst?plain=1#L143-L173

This is the source of truth.  But the function `get_wheel_filename` which is used to reproduce these filenames in the Python code is out of sync with the truth:

https://github.com/ray-project/ray/blob/bb892a6bd0224ab3ae11859268ddef01cbb68e71/python/ray/_private/utils.py#L1384-L1435

- The code produces filenames with `universal` in certain cases, but there are no longer any such filenames
- The code says there aren't master wheels for M1 macs, but this is no longer true
- Possibly other issues

The downstream effect is that the "conda" field of runtime_env won't work on the nightly build in these cases.

We should fix the code to put it back in sync, and add a comment to `installation.rst` that links back to `get_wheel_filename` to help prevent them going out of sync again.

### PR DESCRIPTION
**Pull request by lejara**
## Why are these changes needed?

The `get_wheel_filename` method was not returning the proper filename of the current wheels.
This pr fixes this issue

## Related issue number

Fixes #39842

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_runtime_env.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def test_get_wheel_filename():
     for sys_platform in ["darwin", "linux", "win32"]:
         for py_version in ray_constants.RUNTIME_ENV_CONDA_PY_VERSIONS:
             # TODO(https://github.com/ray-project/ray/issues/31362)
-            if py_version == (3, 11) and sys_platform != "linux":
+            if py_version == (3) and sys_platform != "linux":
```
Comment (architkulkarni) [inline python/ray/tests/test_runtime_env.py:118]: What's the reason for this change?

**Review by architkulkarni — COMMENTED** [review]
Let's update `test_get_wheel_filename` to go through all the possibilities.

**Comment by lejara** [comment]
sure thing

File: python/ray/tests/test_runtime_env.py (hunk 1 above)
Comment (lejara) [reply python/ray/tests/test_runtime_env.py]: I saw you have a wheel for 3.11 but not for 3.3. So thought to only remove the 11. but ill just remove the whole case

Comment (architkulkarni) [reply python/ray/tests/test_runtime_env.py]: Oh sorry, (3,11) refers to 3.11

**Review by architkulkarni — COMMENTED** [review]
Ah looks like there's a test failure because the wheels we're testing weren't around back in Ray 2.5.0, could you update `test_get_release_wheel_url` to have `test_commits = {"2.7.0": "904dbce085bc542b93fbe06d75f3b02a65d3a2b4"}`?

**Comment by architkulkarni** [comment]
And use `"904dbce085bc542b93fbe06d75f3b02a65d3a2b4"` in `test_get_master_wheel_url` as well

**Comment by lejara** [comment]
sure thing. i noticed they both use the same edge case
```python
            # TODO(https://github.com/ray-project/ray/issues/31362)
            if py_version == (3, 11) and sys_platform != "linux":
                continue
```

shall i remove those as well?
And should they go through all the possibilities like the changed made to `test_get_wheel_filename`

**Comment by lejara** [comment]
Another unrelated note.

You mind add the  `hacktoberfest-accepted` label to PR this pr.

Apologies, i could sworn i thought this repo was in hacktoberfest. it showed up on my search results.

If its too much of a ask. its okay!

**Comment by architkulkarni** [comment]
> sure thing. i noticed they both use the same edge case
> 
> ```python
>             # TODO(https://github.com/ray-project/ray/issues/31362)
>             if py_version == (3, 11) and sys_platform != "linux":
>                 continue
> ```
> 
> shall i remove those as well? And should they go through all the possibilities like the changes made to `test_get_wheel_filename`

Yes, that would be ideal, thanks!

**Comment by architkulkarni** [comment]
> And use `"904dbce085bc542b93fbe06d75f3b02a65d3a2b4"` in `test_get_master_wheel_url` as well
Sorry, can you use `0910639b6eba1b77b9a36b9f3350c5aa274578dd` for `test_get_master_wheel_url` instead? I just checked and this one has all the wheels uploaded.

Also, the lint job is failing in buildkite/premerge. You can run `ci/setup_hooks.sh` to install a hook that automatically runs the linter locally before pushing.

**Review by architkulkarni — APPROVED** [review]
Looks good, if you see that the tests are done and the failed tests are unrelated, let me know and I can merge it.

**Comment by lejara** [comment]
> Looks good, if you see that the tests are done and the failed tests are unrelated, let me know and I can merge it.

Alrighty,
of what i can see in the test logs. It seems good 👍

**Comment by angelinalg** [comment]
@architkulkarni Who is the audience for the doc updates? Users of Ray? If so, are they the ones that are changing the wheels, or is the Ray Team changing the wheels. I'm trying to figure out how to use active voice for the sentence, vs passive voice.

**Comment by architkulkarni** [comment]
@angelinalg it's for the ray team updating the wheels. The comment is not visible to users (it's like a code comment)

File: doc/source/ray-overview/installation.rst [hunk 2]
```diff
@@ -138,6 +138,8 @@ You can install the nightly Ray wheels via the following links. These daily rele
 
 .. note::
 
+  .. If changes are made to the list of wheel links below, ensure `get_wheel_filename()` in  `https://github.com/ray-project/ray/blob/master/python/ray/_private/utils.py` are up to date.
```
Comment (angelinalg) [inline doc/source/ray-overview/installation.rst:141]:
```suggestion
  .. If you change the list of wheel links below, remember to update `get_wheel_filename()` in  `https://github.com/ray-project/ray/blob/master/python/ray/_private/utils.py`.
```

**Review by angelinalg — APPROVED** [review]
